File: Java/algoritmo5.java
import java.util.*;

public class algoritmo5 {

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        // Inserir a nota e o peso da nota
        System.out.print("Digite a primeira nota: ");
        double nota1 = scanner.nextDouble();
        System.out.print("Digite o peso da primeira nota: ");
        double peso1 = scanner.nextDouble();
        
        System.out.print("Digite a segunda nota: ");
        double nota2 = scanner.nextDouble();
        System.out.print("Digite o peso da segunda nota: ");
        double peso2 = scanner.nextDouble();
        
        System.out.print("Digite a terceira nota: ");
        double nota3 = scanner.nextDouble();
        System.out.print("Digite o peso da terceira nota: ");
        double peso3 = scanner.nextDouble();

        // Calcula a média ponderada
        double mediaPonderada = (nota1 * peso1 + nota2 * peso2 + nota3 * peso3) / (peso1 + peso2 + peso3);
        System.out.printf("A média ponderada é: %.2f%n", mediaPonderada);

        scanner.close();
    }
}

File: Haskell/algoritmo5.hs
main :: IO ()
main = do
    putStrLn "Digite a primeira nota: "
    nota1 <- readLn :: IO Double
    putStrLn "Digite o peso da primeira nota: "
    peso1 <- readLn :: IO Double

    putStrLn "Digite a segunda nota: "
    nota2 <- readLn :: IO Double
    putStrLn "Digite o peso da segunda nota: "
    peso2 <- readLn :: IO Double

    putStrLn "Digite a terceira nota: "
    nota3 <- readLn :: IO Double
    putStrLn "Digite o peso da terceira nota: "
    peso3 <- readLn :: IO Double

    let mediaPonderada = (nota1 * peso1 + nota2 * peso2 + nota3 * peso3) / (peso1 + peso2 + peso3)

    putStrLn $ "A média ponderada é: " ++ show mediaPonderada
